adafruit_esp32spi/adafruit_esp32spi.py

```diff
@@ -779,6 +786,37 @@ def set_analog_write(self, pin, analog_value):
         if resp[0][0] != 1:
             raise RuntimeError("Failed to write to pin")
 
+    def set_digital_read(self, pin):
```

Comment:
You may want to assert `firmware_version()` => 1.5.0 for `set_digital_read` and `set_analog_read` and raise an error if it's lower. These two methods won't work if your firmware doesn't have these handlers.

Reply:
Same is true for all methods added since factory 1.2.2: enterprise, ntp.

Being a CP n00b, I'll have to look into how to best do the comparison to catch stuff like 1.10.0 > 1.5.0. Or... open to suggestions.

Reply:
Take a look at how AWS IoT performs this check:
https://github.com/adafruit/Adafruit_CircuitPython_AWS_IOT/blob/master/adafruit_aws_iot.py#L70

Reply:
Thanks! Done, and tested against 1.4.0 vs. 1.5.0.